### PR TITLE
fix the ignored tests in doc test

### PR DESCRIPTION
**Pull request by guanqun**
I would like to put the long running tests into ignored tests. these three take too long... It speeds up my dev and testing process.

Now the ci would run both ignored and non ignored ones.

**EDIT**: after discussion, I'll just fix the ignored tests in doc test, move them to either `text` or `nocompile`. After this fix, the `cargo test -- --ignored` can run properly.

### CODE REVIEW

File: ci/script.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ case $TARGET in
 		sudo apt-get -y update
 		sudo apt-get install -y cmake pkg-config libssl-dev
 
-		cargo test --all --locked
+		cargo test --all --locked && cargo test --all --locked -- --ignored
```
Comment (tomusdrw) [inline ci/script.sh]: Running these tests in `--release` makes them a bit faster - maybe something to consider for the CI as well?

Comment (guanqun) [reply ci/script.sh]: For CI I think it's ok to be a bit slow. However the speed is more important for local development.

**Comment by guanqun** [comment]
Another way to do it is to create a new feature gate and put long running tests inside this feature gate.

**Comment by bkchr** [comment]
I once had errors, that just showed up in the long running tests. So, I'm not sure if we should disable them. 
(Currently I just let these tests running for a little bit and cancel them afterwards as they does not finish on my Laptop..(or maybe finish somewhere in a distant future))

**Comment by guanqun** [comment]
We're not disabling them,just moving them to ignored, and these ignored tests are still checked in CI. And users can decide if they want to run the long running tests.

File: ci/script.sh (hunk 1 above)
Comment (gavofyork) [reply ci/script.sh]: there's no point running debug in CI and sometimes CI can be time critical so i'd prefer `--release` here

**Comment by guanqun** [comment]
interesting, the `--release` runs faster while the debug version runs minutes longer... so with the release version, I'm fine with the current behavior, but still I'd like to fix the ignored tests in doc tests. Please help review. thanks.

**Comment by guanqun** [comment]
Adding `--release` to CI script.

**Pull request merged**